Fix for older libavutil + add travis configuration

jogu opened the pull request:
I've tested this with the debian stable libavutil, plus the older one on travis, both work fine.

The travis config means we can setup travis to check that pull requests compile - I'll do that once this pull request is merged.

Mr-Dave commented:
On the ffmpeg change, I was thinking a better approach would be to go with a revision to how the other changes associated with 12.04 were implemented.  Basically create a "MyPixelFormat" enum that embeds the defines [here.](https://github.com/Motion-Project/motion/blob/master/ffmpeg.h#L16).  I should have probably put this as a enum instead of define to start with.

jogu commented:
Hm, I've done a different fix that I thought was consistent with what you said, but looking back at your comment now I'm not sure this is what you meant. I think this may be better - it seems a good idea to define what we use to the actual define that ffmpeg uses. What do you think?

jogu commented:
(I've redone the testing btw, this one works on debian stable & with the old libavutil.)

mterzo commented:
Awesome.

Mr-Dave commented:
I do like this defining better.  I was going to just create a new enum with just the two formats we deal with (defined immediately above.)  In a later pull, the two #if blocks should be consolidated since they are looking for the same change to the ffmpeg api.

jogu commented:
This seems to be working, thanks @mterzo !

I've added a "build: passing" (or failing, updated as appropriate by travis) badge on https://motion-project.github.io.

New pull requests will also automatically be tested by travis, see the "All checks have passed" bit on https://github.com/Motion-Project/motion/pull/158 as an example.

It's testing on the rather old ubuntu 12.04; I noticed travis do have a 14.04 image of some kind too, which I presume has a much newer ffmpeg: https://docs.travis-ci.com/user/trusty-ci-environment/

Might be worth us trying to setup the tests so they run on both 12.04 and 14.04 at some point.

mterzo commented:
Yeah. Seems to be a lot of over head in their 14.04 environment per their docs.
